ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh

```
@@ -297,16 +297,94 @@ echo "Windows OS image ID: $windows_os_image_id"
+echo "DEBUG: WMCO operator logs before wait (last 50 lines):"
+oc logs -n openshift-windows-machine-config-operator deployment/windows-machine-config-operator --tail=50 || echo "Failed to get WMCO logs"
+
+echo "DEBUG: Events in openshift-machine-api namespace:"
+oc get events -n openshift-machine-api --sort-by='.lastTimestamp' | tail -30
+
 echo "Waiting for ${winworker_machineset_replicas} Windows machines to become ready..."
-while [[ $(oc -n openshift-machine-api get machineset/${winworker_machineset_name} -o 'jsonpath={.status.readyReplicas}') != "${winworker_machineset_replicas}" ]]; do echo -n "." && sleep 10; done
+debug_interval=0
+max_wait=2400
+elapsed=0
+while [[ $(oc -n openshift-machine-api get machineset/${winworker_machineset_name} -o 'jsonpath={.status.readyReplicas}') != "${winworker_machineset_replicas}" ]]; do
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)